### PR TITLE
Home View Updates

### PR DESCRIPTION
**Pull request by MohamedSaiko**
Simulator Screen Shot - iPhone 13 Pro - 2024-02-13 at 11 40 45
Requesting code review.

### CODE REVIEW

File: iStore/Home/HomeView.swift [hunk 1]
```diff
@@ -17,8 +17,15 @@ struct HomeView: View {
         NavigationView {
             VStack() {
                 LocationButton()
-                
-                ProductsGrid(products: homeViewModel.products)
+                if homeViewModel.isloading {
```
Comment (akabel92) [inline iStore/Home/HomeView.swift:20]: Why do you use `homeViewModel` as state not environment  object?

File: iStore/Home/ProductsGrid.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+//
+//  Created by Mohamed Sayed on 09.02.24.
+//
+
+import SwiftUI
+
+struct ProductsGrid: View {
+    
+    var products: [Product]
+    private let adaptiveColumn = [GridItem(.adaptive(minimum: 150, maximum: 180))]
```
Comment (akabel92) [inline iStore/Home/ProductsGrid.swift:13]: Why 150 and 100?

File: iStore/Home/ProductsGrid.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                        .frame(width: 170, height: 170, alignment: .center)
+                        .padding(5)
+                    }
+                }
+            }
+        }
+    }
+}
+
+struct ProductsGrid_Previews: PreviewProvider {
```
Comment (akabel92) [inline iStore/Home/ProductsGrid.swift:58]: Why is the reason for adding `ProductsGrid_Previews` here?

**Comment by akabel92** [comment]
Could you add screenshot for the implemented view in the PR description please?

File: iStore/Home/ProductsGrid.swift (hunk 3 above)
Comment (MohamedSaiko) [reply iStore/Home/ProductsGrid.swift:58]: removed it

File: iStore/Home/ProductsGrid.swift (hunk 2 above)
Comment (MohamedSaiko) [reply iStore/Home/ProductsGrid.swift:13]: UI design perspective

File: iStore/Home/HomeView.swift (hunk 1 above)
Comment (MohamedSaiko) [reply iStore/Home/HomeView.swift:20]: - To state the HomeView ownership of the homeViewModel.
- I'm using the homeViewModel object only in the HomeView.
- I only read the model data without writing it.
- I would use the Environment object if I use the object in multiple views.

**Review by akabel92 — APPROVED** [review]

**Pull request merged**